tests/functional/base_metric_types/test_base_average_metric.py

```diff
@@ -0,0 +1,107 @@
+from struct import pack
+import os
+import pytest
+from dbt.tests.util import run_dbt
+
+# our file contents
+from tests.functional.fixtures import (
+    fact_orders_source_csv,
+    fact_orders_sql,
+    fact_orders_yml,
+)
+
+# models/base_average_metric.sql
+base_average_metric_sql = """
+select *
+from 
+{{ metrics.calculate(metric('base_average_metric'), 
+    grain='month'
+    )
+}}
+"""
+
+# models/base_average_metric.yml
+base_average_metric_yml = """
+version: 2 
+models:
+  - name: base_average_metric
+    tests: 
+      - dbt_utils.equality:
+          compare_model: ref('base_average_metric__expected')
+metrics:
+  - name: base_average_metric
+    model: ref('fact_orders')
+    label: Total Discount ($)
+    timestamp: order_date
+    time_grains: [day, week, month]
+    type: average
+    sql: discount_total
+    dimensions:
+      - had_discount
+      - order_country
+"""
+
+# seeds/base_average_metric__expected.csv
+base_average_metric__expected_csv = """
+date_month,base_average_metric
+2022-01-01,1.00000000000000000000
+2022-02-01,1.3333333333333333
+""".lstrip()
+
+class TestBaseAverageMetric:
+
+    # configuration in dbt_project.yml
+    @pytest.fixture(scope="class")
+    def project_config_update(self):
+        return {
+          "name": "example",
+          "models": {"+materialized": "view"}
+        }
+
+    # install current repo as package
+    @pytest.fixture(scope="class")
+    def packages(self):
+        return {
+            "packages": [
+                {"local": os.getcwd()}
+                ]
+        }
+
+
+    # everything that goes in the "seeds" directory
+    @pytest.fixture(scope="class")
+    def seeds(self):
+        return {
+            "fact_orders_source.csv": fact_orders_source_csv,
+            "base_average_metric__expected.csv": base_average_metric__expected_csv,
+        }
+
+    # everything that goes in the "models" directory
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {
+            "fact_orders.sql": fact_orders_sql,
+            "fact_orders.yml": fact_orders_yml,
+            "base_average_metric.sql": base_average_metric_sql,
+            "base_average_metric.yml": base_average_metric_yml
+        }
+
+    def test_build_completion(self,project,):
+        # running deps to install package
+        results = run_dbt(["deps"])
+
+        # seed seeds
+        results = run_dbt(["seed"])
+        assert len(results) == 2
+
+        # initial run
+        results = run_dbt(["run"])
+        assert len(results) == 3
+
+        # test tests
+        results = run_dbt(["test"]) # expect passing test
+        assert len(results) == 1
```

Comment:
What is the `results` object here? Is it an array of passing tests? I would sorta prefer that it returned all results, and a pass/fail, and then we could do whatever the python version of `results.all(result => result.statuscode == 'PASS')` is

Reply:
Because otherwise anytime we add new tests, we have to go through and update the expected number of passes. (or, worse, we add a test which fails, we forget to update the expected count, and it sits there failing forever).

Reply:
@dbeatty10 would have more context around this area. I suspect it returns the results of each command and then we confirm that it matches what we expect. So if we added a new test we'd want to assert that only two tests were running

Reply:
The example containing hard-coded numbers comes from [here](https://docs.getdbt.com/docs/contributing/testing-a-new-adapter#example-a-simple-test-case).

`dbt_metrics` doesn't need to do it that way if it doesn't want to. The important part is to verify that it works as expected for all the possible edge cases. And that it fails in the ways expected with bad input.